handlers/guest_links_test.go

```diff
@@ -347,3 +346,164 @@ func TestDeleteInvalidGuestLink(t *testing.T) {
+		},
+		{
+			description: "disabling guest link succeeds but has no effect when guest link is not disabled",
+			url:         "/api/guest-links/abcdefgh23456789/disable",
+			guestLinkInStore: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     true,
+			},
+			expected: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     true,
+			},
+			status: http.StatusNoContent,
+		},
+		{
+			description: "enabling guest link succeeds but has no effect when guest link is not enabled",
+			url:         "/api/guest-links/abcdefgh23456789/enable",
+			guestLinkInStore: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     false,
+			},
+			expected: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     false,
+			},
+			status: http.StatusNoContent,
+		},
+		{
+			description: "disable a nonexistent guest link",
+			url:         "/api/guest-links/abcdefgh23456789/disable",
+			status:      http.StatusNotFound,
+		}, {
+			description: "enable a nonexistent guest link",
+			url:         "/api/guest-links/abcdefgh23456789/enable",
+			status:      http.StatusNotFound,
+		},
+		{
+			description: "disable a invalid guest link",
+			url:         "/api/guest-links/i-am-an-invalid-link/disable",
+			status:      http.StatusBadRequest,
+		},
+		{
+			description: "enable a invalid guest link",
+			url:         "/api/guest-links/i-am-an-invalid-link/enable",
+			status:      http.StatusBadRequest,
+		},
+	} {
+		t.Run(tt.description, func(t *testing.T) {
+			dataStore := test_sqlite.New()
+
+			if tt.status == http.StatusNoContent {
```

Comment:
Can we change this to `if !tt.guestLinkInStore.Empty() {`?

I had trouble understand at first why we'd condition based on the HTTP response code when we hadn't even sent the HTTP request. I think checking based on whether the guest link is populated is the more direct way of testing whether we should add it into the mock store.